src/collapse/mod.rs

```diff
@@ -58,4 +61,11 @@ pub trait Collapse {
             }
         }
     }
+
+    /// Returns whether this implementation is appropriate for the given input.
+    ///
+    /// None means "not sure -- need more input".
+    /// Some(true) means "yes, this implementation should work with this string".
+    /// Some(false) means "no, this implementation definitely won't work".
```

Comment:
Let's make this a list with ` - ` and add backticks around `None` and `Some(..)`